src/helpers.php

```diff
@@ -8,11 +8,12 @@
     /**
      * @param \Spatie\Async\Task|callable $task
      *
+     * @param string $binary
      * @return \Spatie\Async\Process\ParallelProcess
      */
-    function async($task): Runnable
+    function async($task, string $binary = 'php'): Runnable
```

Comment:
I don't think we should support another binary directly on the `async` function, but rather support it only on the pool:

```php
Pool::create()->withBinary(…);

$pool[] = async();
```

What's the reason you also wanted to support it in `async()`?

Reply:
Because if we add to pool task with async() function, than we can't change binary

Reply:
Right, but if you'd want another binary, it would already be configured on the pool itself, no? I can hardly imagine any scenario where you'd want to use different binaries within the same pool?

Reply:
The thing is that if we create a Task through the async() function, we create a process with a default binary and we cannot change it.

Reply:
I would add a `withBinary` method on the `Runnable` interface. Whenever a runnable is added to the pool (https://github.com/spatie/async/blob/master/src/Pool.php#L116), the pool will call it. 

In practice:

```php
        $process = $process->withBinary($this->binary);

        $this->putInQueue($process);

        return $process;
```

We want to keep the `async` function as clean as possible. 

Does that work for you?

Reply:
Should i try to do it, or i revert changes in async() functions and we merge this pr?

Reply:
If you're up for it, you could revert the changes to `async` and add my suggestion.

Reply:
Can you look at the changes and comment them? In order to implement what we talked about, I had to change the implementation of the Runnable interface slightly.